akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolGateway.scala

```diff
@@ -39,56 +41,74 @@ private object PoolGateway {
  * get reused will automatically forward requests directed at them to the latest pool incarnation from the cache.
  */
 private[http] class PoolGateway(hcps: HostConnectionPoolSetup,
-                                _shutdownStartedPromise: Promise[Done])( // constructor arg only
+                                _shutdownStartedPromise: Option[Promise[Done]] = None)( // used for cache invalidation
                                   implicit system: ActorSystem, fm: Materializer) {
   import PoolGateway._
   import fm.executionContext
 
-  private val state = {
+  private[this] var state: State = {
     val shutdownCompletedPromise = Promise[Done]()
     val props = Props(new PoolInterfaceActor(hcps, shutdownCompletedPromise, this)).withDeploy(Deploy.local)
```

Comment:
This leaks `this` to another thread ( the actor's thread) before fully constructed.